### DISCUSSION TITLE
How does this differ from org-link minor mode?

### DISCUSSION THREAD

**Discussion by mohkale** [tarsius/orglink#10] [Support] [answered]
I recently saw magit added `git-commit-turn-on-orglink` which references this project and since I'm already using org-link minor mode I'm curious how this differs from that.

**Comment by tarsius** [comment, marked as answer]
That brings back some unpleasant memories.

When I looked for such functionality I found `org-link` and used that for a while, even contributing a few minor fixes. But I was not satisfied and discovered that all the good parts had been copied from Org while all the bad parts had been added by Sean.

I decided that the most productive way forward was to start over, copying code from Org and then filling in the missing pieces from scratch. I added Org's copyright notice, acknowledging that I had copied code from there. I did not do the same for Sean's `org-link` because I had not actually copied anything from there. I considered mentioning `org-link` as an alternative implementation but decided against doing so because that would have meant sharing my negative opinion about it, which I felt would not be a nice thing to do and was also unnecessary.

Some time later I got a very angry mail from Sean accusing me of ripping him of and having violated the GPL and demanding that I immediately repent and acknowledged that my package derived from his. This mail could have been written along the lines "I noticed you forgot to mention my copyright, please fix that", and we could have had a constructive conversation about it. Instead it was written in a way that made me feel like I was being accused of being a bad human being. Needless to say, I did not take this lightly.

Instead of merely stating that my implementation does not derive from his, I went through both of them comparing in detail how I had implemented a certain feature, how he did it, and why his approach was very bad. Essentially I was saying that whenever he had a decision to make, he consistently made a bad choice. Had he approached me respectfully, assuming good faith, I would have worded it differently but he had personally attacked me in our very first communication and so I didn't hold back. In retrospective I could and should have responded in a more constructive way, but I failed to do so.

Additionally, and this is more important, I pointed out that both he and I had copied code from Org, but that while I had properly attributed that source with a copyright notice, he had failed to do the same.

I never heard back from him and to this day he has not added a copyright notice that acknowledges that he had copied code from Org. Whether my implementation derives from his is debatable but the fact that both of them derive from Org is absolutely undeniable. It is clear that he feels very differently about violating someone else's copyright, then about a perceived violation of his own rights.

This must be one of, if not the most hypocritical and vicious attacks I ever was the victim of, and for that reason I do not wish to interact with Sean ever again, so please do not bring this statement to his attention.

**Reply by mohkale** [reply]
@tarsius 

I see, I'm sorry for bringing up such painful memories. Thanks for the answer and explanation of why this project exists. I think I'll try it out first hand and maybe replace org-link-minor-mode with it after that. Thank you for all your work and continued dedication on this project and all the other projects you've shared with the community. :+1:.

**Reply by tarsius** [reply]
You're welcome!
